compiler/rustc_typeck/src/check/coercion.rs

```diff
@@ -87,6 +89,29 @@ impl<'a, 'tcx> Deref for Coerce<'a, 'tcx> {
 
 type CoerceResult<'tcx> = InferResult<'tcx, (Vec<Adjustment<'tcx>>, Ty<'tcx>)>;
 
+struct CollectRetsVisitor<'tcx> {
+    ret_vec: Vec<&'tcx hir::Expr<'tcx>>,
+}
+
+impl<'tcx> Visitor<'tcx> for CollectRetsVisitor<'tcx> {
+    fn visit_block(&mut self, b: &'tcx Block<'tcx>) {
+        for stmt in b.stmts {
+            //self.visit_stmt(stmt);
+            intravisit::walk_stmt(self, stmt);
+        }
+    }
```

Comment:
I think that this isn't needed because `Visitor::visit_block` should have a default impl that does the same thing as this (visit every statement, and its tail expression as well).